notebooks/06a Modelling - classification.ipynb

```diff
@@ -0,0 +1,865 @@
+    "* Explore fairness (next notebook) of best performing model\n",
+    "\n",
+    "Logistic regression models are excluded due to significant multi-collinearity (see analysis in Regression notebook)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "gather": {
+     "logged": 1653046713645
+    }
+   },
+   "outputs": [],
+   "source": [
+    "# Initiate empty models dictionary\n",
+    "models = {}"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Prior model\n",
+    "\n",
+    "The simplest baseline model takes the most frequent class label as its prediction"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "gather": {
+     "logged": 1653046716917
+    }
+   },
+   "outputs": [],
+   "source": [
+    "model_name = \"prior\"\n",
+    "\n",
+    "# define gridsearch parameters\n",
+    "gsc = GridSearchCV(\n",
+    "    estimator=DummyClassifier(strategy=\"prior\"),\n",
+    "    param_grid={},\n",
+    "    cv=5,\n",
+    "    scoring=\"f1_weighted\",\n",
+    "    verbose=1,\n",
+    "    n_jobs=-1,\n",
+    "    refit=True,\n",
+    ")\n",
+    "\n",
+    "# takes ~1 second to run on a STANDARD_DS3_V2\n",
+    "models[model_name] = train_model(gsc, X_train, y_train)\n",
+    "models[model_name]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Decision tree classifier (weighted)\n",
+    "\n",
+    "Simplest tree classifier using one tree"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "gather": {
+     "logged": 1653046754064
+    }
+   },
+   "outputs": [],
+   "source": [
+    "model_name = \"decisiontree\"\n",
+    "\n",
+    "# define gridsearch parameters\n",
+    "gsc = GridSearchCV(\n",
+    "    estimator=DecisionTreeClassifier(class_weight=weights.to_dict()),\n",
+    "    param_grid={\"max_depth\": [5, 10, None]},  # default None\n",
+    "    cv=5,\n",
+    "    scoring=\"f1_weighted\",\n",
+    "    verbose=1,\n",
+    "    n_jobs=-1,\n",
+    "    refit=True,\n",
+    ")\n",
+    "\n",
+    "# takes ~10s to run on a STANDARD_DS3_V2\n",
+    "models[model_name] = train_model(gsc, X_train, y_train)\n",
+    "models[model_name]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Random forest (weighted)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "gather": {
+     "logged": 1653046912982
+    }
+   },
+   "outputs": [],
+   "source": [
+    "model_name = \"randomforest\"\n",
+    "\n",
+    "# define gridsearch parameters\n",
+    "gsc = GridSearchCV(\n",
+    "    estimator=RandomForestClassifier(class_weight=weights.to_dict()),\n",
+    "    param_grid={\n",
+    "        \"n_estimators\": [50, 100, 200],  # default 100\n",
+    "        \"max_depth\": [5, 10, None],\n",
```

Comment:
Having no max depth here could lead to overfitting as with Decision Tree (perhaps less of an issue here due to nature of Random Forest)